integration_tests_quick_start_guide.rst

```diff
@@ -0,0 +1,646 @@
+2. Add a task which reproduces the to an appropriate file within the ``tests/integration/targets/<target_name>/tasks`` directory.
+3. :ref:`Run the tests<Run-integration-tests>`, they should fail.
+4. If they do not fail, re-check if your environment / test task satisfies the conditions described in the ``Steps to Reproduce`` section of the issue.
+5. If you reproduce the bug and tests fail, change the code. 
+6. :ref:`Run the tests<Run-integration-tests>` again.
+7. If they fail, repeat steps 5-6 until the tests pass.
+
+Here is an example.
+
+Let's say we got an issue in the ``community.postgresql`` collection. When users pass a name containing underscores to the ``postgresql_user`` module, the module fails.
+
+We cloned the collection repository to the ``~/ansible_collections/community/postgresql`` directory and :ref:`prepared our environment <Prepare-local-environment>`. Being there, we run ``ansible-test integration --list-targets`` and it shows a target called ``postgresql_user``. It means that we already have tests for the module.
+
+We start with reproducing the bug.
+
+First, we look into the ``tests/integration/targets/<target_name>/tasks/main.yml`` file. In case of the ``community.postgresql``, it imports other files from the ``tasks`` directory. We looked through the files - ``postgresql_user_general.yml`` looks like an appropriate one to add our tests.
+
+.. code:: yaml
+
+  # General tests:
+  - import_tasks: postgresql_user_general.yml
+    when: postgres_version_resp.stdout is version('9.4', '>=')
+
+We will add the following code to the file.
+
+.. code:: yaml
+
+  # https://github.com/ansible-collections/community.postgresql/issues/NUM
+  - name: Test user name containing underscore
+    postgresql_user:
+      name: underscored_user
+    register: result
+
+  - name: Check the module returns what we expect
+    assert:
+      that:
+        - result is changed
+
+  - name: Query the database if the user exists
+    postgresql_query:
+      query: SELECT * FROM pg_authid WHERE rolename = 'underscored_user'
+    register: result
+
+  - name: Check the database returns one row
+    assert:
+      that:
+        - query_result.rowcount == 1
+
+When we :ref:`run the tests<Run-integration-tests>` passing ``postgresql_user`` as a test target, this task must fail.
+
+Then we will fix the bug and run the same tests again. If they pass, we will consider the bug fixed and will submit a pull request.
+
+When adding a new feature
+-------------------------
+
+.. note::
+
+  The process described in this section is also applicable when a feature already exists but does not have integration tests and you want to cover it.
+
+.. note::
+
+  If you don not implement the feature you want yet, you can start with writing integration tests for it. Of course they will not work as the code does not exist at the moment but it can help you design better implementation before writing the code.
```

Comment:
```suggestion
  If you do not implement the feature you want yet, you can start with writing integration tests for it. Of course they will not work as the code does not exist at the moment but it can help you design better implementation before writing the code.
```